schema/sql/api.sql

```diff
@@ -31,6 +31,17 @@ grant usage on sequence "user_id" to "api-user-create";
 grant "api-user-create" to "api";
 grant "api-anon" to "api-user-create";
 
+create role "api-user-update";
+comment on role "api-user-update" is
+    $$Intended for use with user self-management systems$$;
+grant usage on sequence "user_id" to "api-user-create";
+grant select,insert,update,delete on table
+    public."hosts",
+    public."passwd",
+    public."ssh_public_key",
+    public."openpgp_public_key"
```

Comment:
These likely need to get RLS rules

Reply:
I don't know what that means

Reply:
RLS = row level security.
i.e. *which* rows of `openpgp_public_key`/`ssh_public_key` are you allowed to modify?

Which depends on how much you trust the client.
e.g. you might say that you have to have a header `My-Username` set to a matching `passwd` entry to be able to add public keys for it.
Alternatively you may just trust the client to always do the right thing.

Reply:
The client is doing ssh authentication against the user, so I am reasonably confident that I will always know the correct username.

If you know of any nice upgrades here to make it stricter, but not break, I am all ears, else I don't see this as a blocker.

Reply:
Wouldn't this mean that anyone with access to the credentials would be able to change any users information if RLS is not enabled?